### ISSUE TITLE
Change API in jwt-cpp

### ISSUE THREAD

**Issue by neon1ks** [artiomn/cpp-network-tasks#11]
Take note!

In release v0.7.0-rc.0 from Dec 5, 2022:
 
- `get_payload_claims()` was `replaced by get_payload_json()`

https://github.com/Thalhammer/jwt-cpp/releases

When updating the library, the following code is affected:

```diff
diff --git a/src/l6/jwt-cpp-example/main.cpp b/src/l6/jwt-cpp-example/main.cpp
index a2b6c8c..864499e 100644
--- a/src/l6/jwt-cpp-example/main.cpp
+++ b/src/l6/jwt-cpp-example/main.cpp
@@ -20,6 +20,7 @@ int main()
 
     verifier.verify(decoded);
 
-    for(auto& e : decoded.get_payload_claims())
+    // for(auto& e : decoded.get_payload_claims()) // old api
+    for(auto& e : decoded.get_payload_json())
         std::cout << e.first << " = " << e.second << std::endl;
 }
```

**Comment by neon1ks** [comment]
Не увидел макроса, указывающего версию библиотеки jwt-cpp. Поэтому затрудняюсь как подправить код без поломки обратной совместимости.

**Comment by artiomn** [comment]
> Не увидел макроса, указывающего версию библиотеки jwt-cpp. Поэтому затрудняюсь как подправить код без поломки обратной совместимости.

Его нет, похоже: https://github.com/Thalhammer/jwt-cpp/issues/273
Тут важный момент: если оно собирается в контейнере, имеет смысл править.
Я не обновляю GB образ, т.к. не работаю с ними уже, к тому же, не хочется ничего сломать.

В любом случае, проверю и волью, если работает. К тому же, есть образы для книги, там вариант библиотеки новее.

**Comment by neon1ks** [comment]
> К тому же, есть образы для книги, там вариант библиотеки новее.

А что за книга? Есть ли возможность её купить?

**Comment by artiomn** [comment]
Да, конечно. В основу положены курсы. Но там значительно переработан материал и 
Образы уже доступны (начинаются с `nb-`, остальное название такое же, но совместимость не обещаю).
Книга выйдет, похоже, в конце лета, сначала в РФ.
